src/Autofac.Pooling/PoolInstanceContainer.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Autofac.Pooling
+{
+    /// <summary>
+    /// Container class that that will return the wrapped instance to the provided pool when it is disposed.
+    /// </summary>
+    /// <typeparam name="TLimit">The limit type.</typeparam>
+    internal sealed class PoolInstanceContainer<TLimit> : IDisposable
```

Comment:
I might choose a different name (though I don't know what) than `...Container` because we sort of already have a definition for what a "container" is in this domain.

Reply:
Renamed to `PooledInstanceTracker`.